### PR TITLE
Next steps on e2e after creating workload clusters

### PR DESCRIPTION
**Pull request by johnbelamaric**

### CODE REVIEW

**Comment by electrocucaracha** [comment]
@johnbelamaric I just noticed that the instructions mentioned to use an Ubuntu 22.04 instance, eventually this needs to be changed for 20.04 given the kernel [requirements of free5gc](https://github.com/free5gc/gtp5g#notice)

File: e2e/provision/README.md [hunk 1]
```diff
@@ -106,6 +106,11 @@ connection with that setting).
        template:
          annotations:
            approval.nephio.org/policy: initial
+      pipeline:
```
Comment (electrocucaracha) [inline e2e/provision/README.md:109]: I think that the identation is wrong

```console
cat <<EOF | kubectl apply -f -
apiVersion: config.porch.kpt.dev/v1alpha2
kind: PackageVariantSet
metadata:
  name: edge-clusters
spec:
  upstream:
    repo: nephio-example-packages
    package: nephio-workload-cluster
    revision: v5
  targets:
    - repositories:
      - name: mgmt
        packageNames:
          - edge01
          - edge02
          - edge03
          - edge04
      template:
        annotations:
          approval.nephio.org/policy: initial
  pipeline:
    mutators:
      - image: gcr.io/kpt-fn/set-labels:v0.2.0
        configMap:
          nephio.org/site-type: edge
EOF
```

Comment (electrocucaracha) [reply e2e/provision/README.md:109]: BTW, this needs some changes on the CRD

```console
Error from server (BadRequest): error when creating "edge-clusters.yaml": PackageVariantSet in version "v1alpha2" cannot be handled as a PackageVariantSet: strict decoding error: unknown field "spec.pipeline"
```

Comment (johnbelamaric) [reply e2e/provision/README.md:109]: That's the due to the bad indentation

**Comment by johnbelamaric** [comment]
> @johnbelamaric I just noticed that the instructions mentioned to use an Ubuntu 22.04 instance, eventually this needs to be changed for 20.04 given the kernel [requirements of free5gc](https://github.com/free5gc/gtp5g#notice)

Oh, interesting. Because I had some problem when I tried to use 2004, that's why I chose that. I can't remember what that was now.

File: e2e/provision/README.md (hunk 1 above)
Comment (johnbelamaric) [reply e2e/provision/README.md]: Pushed a fix

File: e2e/provision/README.md [hunk 2]
```diff
@@ -118,14 +123,193 @@ connection with that setting).
...
+   three clusters had machines come up:
+
+   ```
+   $ kubectl get machinesets
+   NAME                                   CLUSTER    REPLICAS   READY   AVAILABLE   AGE     VERSION
+   edge01-md-0-6fk8w-9dc5bb56dxnpz69      edge01     3                              3h46m   v1.26.3
+   edge02-md-0-blqfh-57fc564884xjf865     edge02     3          3       3           3h49m   v1.26.3
+   edge03-md-0-8bss6-5b95ddf7b8xf8fg5     edge03     3          3       3           3h48m   v1.26.3
+   edge04-md-0-z7b8p-77748bffbfxwsjbc     edge04     3                              3h46m   v1.26.3
+   test01-md-0-2zkcp-666fb857c6x6jnzk     test01     3          3       3           3h55m   v1.26.3
```
Comment (electrocucaracha) [inline e2e/provision/README.md:156]: Where this test01 machineset came from?

Comment (johnbelamaric) [reply e2e/provision/README.md:156]: Just from other testing. I can remove it from the list to avoid confusion.

Comment (johnbelamaric) [reply e2e/provision/README.md]: Ok, I tweaked the responses of the command to be more in line with what you would see if you did this in the order of the README. I did it in a different order and so some things show up when they shouldn't. Eventually we can recapture command output on a clean run.

File: e2e/provision/README.md [hunk 3]
```diff
@@ -106,6 +106,11 @@ connection with that setting).
        template:
          annotations:
            approval.nephio.org/policy: initial
+         pipeline:
```
Comment (vishwanathj) [inline e2e/provision/README.md:109]: Question: Are L89-L113 captured as a KRM resource in the https://github.com/nephio-project/nephio-example-packages repo or any other repo?
Would it make sense?

File: e2e/provision/README.md [hunk 4]
```diff
@@ -118,14 +123,192 @@ connection with that setting).
...
    $
    ```
+
+   You should also check that the Kind cluster came up fully with `kubectl get
+   machinesets`. Sometimes they do not all come up; it's not clear why yet, but
+   is probably a resourcing issue on the VM. For example, in this case only
+   two clusters had machines come up:
+
+   ```
+   $ kubectl get machinesets
```
Comment (vishwanathj) [inline e2e/provision/README.md:150]: Is this more of a troubleshooting step or verification step?

File: e2e/provision/README.md [hunk 5]
```diff
@@ -118,14 +123,192 @@ connection with that setting).
...
+   NAME                                   CLUSTER    REPLICAS   READY   AVAILABLE   AGE     VERSION
+   edge01-md-0-6fk8w-9dc5bb56dxnpz69      edge01     3                              3h46m   v1.26.3
+   edge02-md-0-blqfh-57fc564884xjf865     edge02     3          3       3           3h49m   v1.26.3
+   edge03-md-0-8bss6-5b95ddf7b8xf8fg5     edge03     3          3       3           3h48m   v1.26.3
+   edge04-md-0-z7b8p-77748bffbfxwsjbc     edge04     3                              3h46m   v1.26.3
+   ```
+
+   If that's the case, pull out the kubeconfig for one of the clusters with
```
Comment (vishwanathj) [inline e2e/provision/README.md:158]: Where would be the location of the kubeconfig files for the different edge clusters?

File: e2e/provision/README.md (hunk 3 above)
Comment (johnbelamaric) [reply e2e/provision/README.md:109]: We could, yeah. Then the workflow would be to manually clone a pkg containing this resource into the mgmt repo, edit the resource for the clusters you want, and publish it.

We could in fact include this and resources in steps 9 and 11 and even more all in one package, and you could deploy everything: clusters, operator, free5gc-cp, and smf/upf/amf all in a single action of publishing one package.

File: e2e/provision/README.md (hunk 4 above)
Comment (johnbelamaric) [reply e2e/provision/README.md:150]: I guess a verification step. We want to make sure that the kind clusters have come up.

File: e2e/provision/README.md (hunk 5 above)
Comment (johnbelamaric) [reply e2e/provision/README.md:158]: The kubeconfigs are all in Secrets "<name>-kubeconfig".

I don't want to include too much explanation/detail in these steps - this is supposed to be a "quick start". We can have a more detailed, blow-by-blow explanation of each step along with troubleshooting tips for each in a separate doc.

**Comment by johnbelamaric** [comment]
> > @johnbelamaric I just noticed that the instructions mentioned to use an Ubuntu 22.04 instance, eventually this needs to be changed for 20.04 given the kernel [requirements of free5gc](https://github.com/free5gc/gtp5g#notice)
> 
> Oh, interesting. Because I had some problem when I tried to use 2004, that's why I chose that. I can't remember what that was now.

So, I tried it with 20.04 now and it worked just fine. Updated the README. Thanks.

**Pull request merged**